### PR TITLE
[SYCL][CUDA] Replace assert with CHECK

### PR DESCRIPTION
**Pull request by bjoernknafla**
Fix LIT tests that are not compiled with assertions enabled but call
assert.

Only done for LIT tests that requires fixes during work on CUDA.

Signed-off-by: Bjoern Knafla <bjoern@codeplay.com>

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
> Only done for LIT tests that requires fixes during work on CUDA.

Anyway, that's a really good thing to do. Thanks!

File: sycl/test/basic_tests/buffer/buffer_full_copy.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-
+#include "../../helpers.hpp"
 #include <CL/sycl.hpp>
 #include <cassert>
```
Comment (bader) [inline sycl/test/basic_tests/buffer/buffer_full_copy.cpp]:
```suggestion

```
Do we still need this include?

File: sycl/test/basic_tests/image_api.cpp [hunk 2]
```diff
@@ -13,9 +13,9 @@
 #include <detail/kernel_impl.hpp>
 #include <detail/scheduler/scheduler.hpp>
 
+#include "../helpers.hpp"
```
Comment (bader) [inline sycl/test/basic_tests/image_api.cpp]: What about replacing `assert` in this file?

File: sycl/test/fpga_tests/fpga_queue.cpp [hunk 3]
```diff
@@ -42,9 +43,10 @@ int getExpectedQueueNumber(cl_device_id device_id, int default_value) {
                                  sizeof(reportedProps),
                                  &reportedProps,
                                  NULL);
-   assert(CL_SUCCESS == iRet && "Failed to obtain queue info from ocl device");
+   CHECK(CL_SUCCESS == iRet && "Failed to obtain queue info from ocl device");
```
Comment (bader) [inline sycl/test/fpga_tests/fpga_queue.cpp]: Why only part of the asserts are replaced in this file?

File: sycl/test/basic_tests/buffer/buffer_full_copy.cpp (hunk 1 above)
Comment (bjoernknafla) [reply sycl/test/basic_tests/buffer/buffer_full_copy.cpp]: It shouldn’t be necessary - I will fix this on Monday.

File: sycl/test/basic_tests/image_api.cpp (hunk 2 above)
Comment (bjoernknafla) [reply sycl/test/basic_tests/image_api.cpp]: This is probably (cannot check at the moment) an artifact of breaking a far too large for its own good PR into smaller PRs. I will check if that is the case and repair it.

File: sycl/test/fpga_tests/fpga_queue.cpp (hunk 3 above)
Comment (bjoernknafla) [reply sycl/test/fpga_tests/fpga_queue.cpp]: Thank you for catching this. Breaking a larger PR down into smaller ones I might have lost some changes (or I was sloppy in the first place). I will look into it.

File: sycl/test/basic_tests/info.cpp [hunk 4]
```diff
@@ -11,6 +11,7 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+#include "../helpers.hpp"
```
Comment (romanovvlad) [inline sycl/test/basic_tests/info.cpp]: Consider adding dir to the including directories in lit.cfg.py.

Comment (romanovvlad) [reply sycl/test/basic_tests/info.cpp]: @bjoernknafla Will you do it or we merging as is?

Comment (bjoernknafla) [reply sycl/test/basic_tests/info.cpp]: I am currently experiencing many LIT failures on the `sycl` branch and am investigating before I can rebase.

As @bader pointed out in another PR, also there is no need to replace `assert` as it isn't turned off when not explicitly defining the `NDEBUG` preprocessor symbol. I was confused as CMake has extra behavior but these LIT tests do not go through CMake. I am considering rejecting and closing this PR.

Comment (bader) [reply sycl/test/basic_tests/info.cpp]: Sounds good to me.

**Comment by bjoernknafla** [comment]
As @bader  pointed out here: https://github.com/intel/llvm/pull/1304#issuecomment-599457235 - it is compeltely fine to use `assert` as no explicit `NDEBUG` is set by the LIT tests. Therefore I will close this PR.

**Pull request closed without merging**